apple/internal/ios_rules.bzl

```diff
@@ -899,6 +903,183 @@ def _ios_extension_impl(ctx):
         ),
     ] + processor_result.providers
 
+def _ios_dynamic_framework_impl(ctx):
+    """Experimental implementation of ios_dynamic_framework."""
+    
+    deps_list = [deps for deps in ctx.attr.deps]
+    binary_target = deps_list.pop()
```

Comment:
it's confusing to me that the ordering of the targets in `deps` is significant to analysis, and could potentially cause issues because `buildifier` automatically sorts `deps`

Reply:
Good call. This was copied from an early version of ios_static_framework_impl. I'll update it to match the current implementation and just take the first dep. Just like for the ios_static_framework, this rule only supports a single direct swift_library target, so taking the first dep should be fine

Reply:
So it turns out there are two deps, even when you only pass in one swift_library as the only dependency. The dep we really need is the swift_runtime_linkopts because that is the target where the SwiftDynamicFrameworkInfo provider gets stored. I updated the statement so that it will search the deps and select the correct one, instead of depending on the order of the deps_list.